runtime/compiler/runtime/JITServerAOTDeserializer.cpp

```diff
@@ -640,6 +642,28 @@ JITServerAOTDeserializer::cacheRecord(const WellKnownClassesSerializationRecord
    }
 
 
+bool
+JITServerAOTDeserializer::cacheRecord(const ThunkSerializationRecord *record,
+                                      TR::Compilation *comp, bool &isNew, bool &wasReset)
+   {
```

Comment:
I'll have to think deeper if this method needs a lock like all the others. Maybe the SCC can handle the concurrency aspect

Reply:
`isResetInProgress()` technically needs a preceding read barrier if we're not acquiring a lock here.